### ISSUE TITLE
Weird issue using FIND_MY_STRUCTURES and STRUCTURE_WALL

### ISSUE THREAD

**Issue by tprk77** [screepers/typed-screeps#85]
### Expected Behavior

I want to write something like the following and have it compile:

```ts
tower.room.find(FIND_MY_STRUCTURES, {
    filter: (st) => st.structureType === STRUCTURE_WALL,
});
```

### Actual Behavior

```ts
// Works fine...
tower.room.find(FIND_STRUCTURES, {
    filter: (st) => st.structureType === STRUCTURE_WALL,
});

// Also works fine...
tower.room.find(FIND_MY_STRUCTURES, {
    filter: (st) => st.structureType === STRUCTURE_RAMPART,
});

// Doesn't compile!
// Error: ~/screeps-typescript-starter/src/main.ts(16,23): semantic error TS2365 Operator '==='
// cannot be applied to types '"extension" | "rampart" | "spawn" | "link" | "storage" | "tower"
// | "observer" | "powerSpawn" | "e...' and '"constructedWall"'.
tower.room.find(FIND_MY_STRUCTURES, {
    filter: (st) => st.structureType === STRUCTURE_WALL,
});`
```

It seems very strange that this doesn't compile, when other very similar code does. So I think this must be a bug? To make sure the problem wasn't with my own project, I cloned `screeps-typescript-starter` and was able to reproduce the problem again.

### Your Environment

- Node.js version: v8.11.3

Basically a newly cloned `screeps-typescript-starter`:

```
"devDependencies": {
    "@types/lodash": "^3.10.1",
    "@types/node": "^10.5.5",
    "@types/screeps": "^2.4.0",
    "prettier": "^1.14.0",
    "rollup": "^0.63.4",
    "rollup-plugin-clear": "^2.0.7",
    "rollup-plugin-commonjs": "^9.1.4",
    "rollup-plugin-node-resolve": "^3.3.0",
    "rollup-plugin-screeps": "^0.1.2",
    "rollup-plugin-typescript2": "^0.16.1",
    "tslint": "^5.9.1",
    "tslint-config-prettier": "^1.14.0",
    "typescript": "^2.9.2"
  },
  "dependencies": {
    "source-map": "~0.6.1"
  }
```

**Comment by semperrabbit** [comment]
I do not use TS, nor am I familiar with the error you are getting, but it would make sense that TS would flip out, as `StructureWall` is not owned so it would not show up in `FIND_MY_STRUCTURES`.

**Comment by tprk77** [comment]
Oh, right! Thank you for pointing out my mistake. That makes sense.

(I guess I thought that I would "own" the wall I built. Sorry, I am still a bit new to Screeps.)

**Comment by apemanzilla** [comment]
Unfortunately we can't make the compiler errors friendlier, but this is the exact type of glitch that TS helps prevent :smiley:
